iml-api/src/graphql/mod.rs

```diff
@@ -1436,6 +1437,50 @@ async fn insert_task(
     Ok(x)
 }
 
+async fn insert_fidlist(fids: Vec<String>, task_id: i32, pool: &PgPool) -> Result<(), ImlApiError> {
+    let x = fids
+        .iter()
+        .map(|fid| LustreFid::from_str(&fid).unwrap())
+        .fold((vec![], vec![], vec![], vec![]), |mut acc, fid| {
+            acc.0.push(fid.seq);
+            acc.1.push(fid.oid);
+            acc.2.push(fid.ver);
+            acc.3.push(serde_json::json!({}));
+
+            acc
+        });
+
+    sqlx::query!(
+        r#"
+            INSERT INTO chroma_core_fidtaskqueue (fid, data, task_id)
+            SELECT row(seq, oid, ver)::lustre_fid, data, $5
+            FROM UNNEST($1::bigint[], $2::int[], $3::int[], $4::jsonb[])
+            AS t(seq, oid, ver, data)"#,
+        &x.0,
+        &x.1,
+        &x.2,
+        &x.3,
+        task_id
+    )
+    .execute(pool)
+    .await?;
```

Comment:
You don't need to create an empty JSON object each iteration and pass it in, you can use a single `jsonb` object instead within the insert statement.

```suggestion
        .fold((vec![], vec![], vec![]), |mut acc, fid| {
            acc.0.push(fid.seq);
            acc.1.push(fid.oid);
            acc.2.push(fid.ver);

            acc
        });

    sqlx::query!(
        r#"
            INSERT INTO chroma_core_fidtaskqueue (fid, data, task_id)
            SELECT row(seq, oid, ver)::lustre_fid, '{}'::jsonb, $4
            FROM UNNEST($1::bigint[], $2::int[], $3::int[])
            AS t(seq, oid, ver)"#,
        &x.0,
        &x.1,
        &x.2,
        task_id
    )
    .execute(pool)
    .await?;
```

Reply:
Done